autoemulate/transforms/pca.py

```diff
@@ -60,6 +60,28 @@ def log_abs_det_jacobian(self, x: TensorLike, y: TensorLike):
         )
         raise RuntimeError(msg)
 
+    def forward_shape(self, shape):
+        """
+        Compute the forward shape transformation.
+
+        For PCA: (batch_shape..., d) -> (batch_shape..., n_components)
```

Comment:
Personally I find it clearer this way that we are maintaining all dimensions except the last one but might just be me

```suggestion
        For PCA: (batch_shape, ..., d) -> (batch_shape, ..., n_components)
```